doc/develop/test/babblesim.rst

```diff
@@ -0,0 +1,484 @@
+``tests/bluetooth/bsim_bt/bsim_test_gatt`` test may look like below:
+
+.. code-block:: yaml
+
+    tests:
+      bluetooth.bsim.gatt:
+        platform_allow:
+          - nrf52_bsim
+        tags:
+          - bluetooth
+        extra_args:
+          - '-DCMAKE_C_FLAGS="-Werror"'
+          - "-DCONFIG_COVERAGE=y"
+          - "-DCMAKE_EXPORT_COMPILE_COMMANDS=ON"
+        bsim_config:
+          devices:
+            -
+              testid: gatt_client
+            -
+              testid: gatt_server
+          phy_layer:
+            name: bs_2G4_phy_v1
+            sim_length: 60e6
+
+This YAML file define necessary information to build and run BabbleSim tests.
+Next after collecting all tests, Pytest filter tests if some filter rules were
+passed in CLI.
+
+Each test consists of two phases:
+
+1.  Building process
+        -   get CMake extra arguments from ``bs_testsuite.yaml`` file and
+            prepare CMake command
+        -   run CMake
+        -   run Ninja generator
+2.  Run simulation
+        -   parse ``bsim_config`` from ``bs_testsuite.yaml`` file and prepare
+            suitable commands to run each simulated devices and physical layer
+        -   run simulation
+        -   if some error/failure occurs during run simulation then mark test as
+            ``FAILED`` - otherwise as ``PASSED``
+
+There is also possibility to generate final report. More information about this
+can be found in chapter `reporting (from Pytest)`_.
+
+Minimal test configuration
+**************************
+
+When test source code is already prepared, then to make it possible to build
+and run them by Pytest, the ``bs_testsuite.yaml`` file have to be defined.
+Let's assume, that exemplary test source is placed in
+``tests\bluetooth\bsim_bt\some_test`` directory. Yaml file have to be defined
+inside them.
+
+Minimal ``bs_testsuite.yaml`` file consists list of test scenarios with
+BabbleSim configs. Each test scenario must have unique name among all test
+scenarios (otherwise Pytest will rise an error). This is very significant,
+because this name will be used to mark BabbleSim simulation ID and to create
+test output directory.
+
+Exemplary basic ``bs_testsuite.yaml`` file could look like:
+
+.. code-block:: yaml
+
+    tests:
+      bluetooth.bsim.some_test:
+        bsim_config:
+          devices:
+            -
+              testid: name_of_first_device
+            -
+              testid: name_of_second_device
+          phy_layer:
+            name: bs_2G4_phy_v1
+            sim_length: 60e6
+
+After running Pytest, it will scan such prepared yaml file and will save
+``bluetooth.bsim.some_test`` test on list of available tests. During test
+execution following CMake command will be called:
+
+::
+
+    cmake -B${ZEPHYR_BASE}/bsim_tests_out/bluetooth_bsim_some_test/build \
+    -S${ZEPHYR_BASE}/tests/bluetooth/bsim_bt/some_test -GNinja \
+    -DBOARD_ROOT=${ZEPHYR_BASE} -DBOARD=nrf52_bsim -DCONF_FILE=prj.conf
```

Comment:
I decided to use `CMake` directly, because the same approach is used in original bash scripts and in Twister. But maybe you are right that, there should be used `west` just to popularize and standardize this building process across whole Zephyr project.